niftilib/nifti1_io.c

```diff
@@ -3652,7 +3656,7 @@ nifti_image* nifti_convert_nhdr2nim(struct nifti_1_header nhdr,
       *   the qform_code will be zero, at which point you can check
       *   analyze75_orient if you care to.
       */
-     unsigned char c = *((char *)(&nhdr.qform_code));
+     unsigned char c = *((unsigned char *)(&nhdr.qform_code));
```

Comment:
This is okay, but does not really matter.  The original field is signed in C, which is probably why the original cast was signed.  If any compilers care about that aspect, this could be better as:
`(unsigned char)*((char *)(&nhdr.qform_code))`
which might have been considered overkill.

Reply:
I've dropped this change.